### PR TITLE
AWS - xpander 1852

### PR DESCRIPTION
**Pull request by johnnywilkes**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-hq.paloaltonetworks.local/browse/EXPANDR-1852

## Description
"AWS - Security Group Remediation v2" playbook is an improvement on V1 because instead of just applying a limited security group, this determines which SG is over-permissive and creates a copy of it without the over-permissive parts.

## Screenshots
N/A

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [X] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Comment by johnnywilkes** [comment]
@yucohen , i shared demo with you

File: Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+        int_sg_mapping (Dict): interface to security group mapping
+        port (int): TCP/UDP Port to be restricted
+        protocol (str): Protocol of the port to be restricted
+
+    Returns:
+        List: list of SGs to be replaced
+    """
+    replace_list = []
+    for mapping in int_sg_mapping.keys():
+        for sg in int_sg_mapping[mapping]:
```
Comment (yucohen) [inline Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py:188]:
```suggestion
    for mapping in int_sg_mapping.values():
        for sg in mapping:
```
This change is optional, but this way the code is easier to read :)

**Review by yucohen — CHANGES_REQUESTED** [review]
@johnnywilkes Great work!!! Please see my comments :) 
@efelmandar Do you think we need a demo for the playbook? Or the linked demo will do?

File: Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+                        group_list.append(sg['GroupId'])
+                    mapping_dict[interface['NetworkInterfaceId']] = group_list
+        if match is False:
+            raise ValueError('could not find interface with public IP association')
+        return mapping_dict
+    else:
+        raise ValueError('failed to pull information on EC2 instance')
+
+
+''' COMMAND FUNCTION '''
```
[the remark this answers is not in the document]
Comment (johnnywilkes) [reply Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py:234]: not a problem, will remove

File: Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+
+
+def aws_recreate_sg_command(args: Dict[str, Any]) -> str:
```
[the remark this answers is not in the document]
Comment (johnnywilkes) [reply Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py:237]: changing to `aws_recreate_sg`

File: Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py (hunk 1 above)
Comment (johnnywilkes) [reply Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py:188]: I made this change and unit tests were breaking.  Will keep as-is

File: Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (johnnywilkes) [reply Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py]: [Yuval's feedback](https://github.com/demisto/content/pull/25339/commits/585c5a6f8cf2f01f48363f7aedc38d46b7022f87)

File: Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+
+    Args:
+        sg_info (Dict): Dictionary of information pulled from "aws-ec2-describe-security-groups" command
+        port (int): TCP/UDP Port to be restricted
+        protocol (str): Protocol of the port to be restricted
+
+    Returns:
+        Dict: Dict of the new SG to be used
+    """
+    info = sg_info[0]['Contents']['AWS.EC2.SecurityGroups(val.GroupId === obj.GroupId)'][0]
```
[the remark this answers is not in the document]
Comment (johnnywilkes) [reply Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py:69]: [Yuval's feedback](https://github.com/demisto/content/pull/25339/commits/585c5a6f8cf2f01f48363f7aedc38d46b7022f87)

File: Packs/AWS-Enrichment-Remediation/README.md [hunk 5]
```diff
@@ -36,4 +42,12 @@ AWS - Unclaimed S3 Bucket Validation playbook validates the unclaimed S3 bucket
...
\ No newline at end of file
+![AWS - Unclaimed S3 Bucket Remediation](https://raw.githubusercontent.com/demisto/content/7f3e223a86964eda5689a9c3e1f22511021c5f40/Packs/AWS-Enrichment-Remediation/doc_files/AWS_-_Unclaimed_S3_Bucket_Remediation.png)
+
+### Automation Scripts
+
+#### AWSRecreateSG
+
+Automation to determine which interface on an EC2 instance has an over-permissive security group, determine which security groups have over-permissive rules, and replace them with a copy of the security group that has only the over-permissive portion removed.  Over-permissive is defined as sensitive ports (SSH, RDP, etc) being exposed to the internet via IPv4.
+
+TODO ADD LINK AFTER PR
```
[the remark this answers is not in the document]
Comment (johnnywilkes) [reply Packs/AWS-Enrichment-Remediation/README.md]: [Yuval's feedback](https://github.com/demisto/content/pull/25339/commits/585c5a6f8cf2f01f48363f7aedc38d46b7022f87)

**Comment by johnnywilkes** [comment]
@ShirleyDenkberg , can you look this over as well, please?

File: Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_Security_Group_Remediation_v2.yml [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "31":
+    id: "31"
+    taskid: e71350f2-0f75-41e1-841d-ee3c938fc23f
+    type: condition
+    task:
+      id: e71350f2-0f75-41e1-841d-ee3c938fc23f
+      version: -1
+      name: Is AWS - EC2 enabled and are Input values defined?
```
Comment (ShirleyDenkberg) [inline Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_Security_Group_Remediation_v2.yml:75]:
```suggestion
      name: Is AWS - EC2 enabled and are input values defined?
```

File: Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_Security_Group_Remediation_v2.yml [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+  required: true
+  description: ID of the AWS EC2 instance.
+  playbookInputQuery:
+- key: Port
+  value:
+    complex:
+      root: alert
+      accessor: remoteport
+  required: true
+  description: TCP/UDP Port to be restricted.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_Security_Group_Remediation_v2.yml:222]:
```suggestion
  description: TCP/UDP port to be restricted.
```

File: Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_Security_Group_Remediation_v2_README.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+This playbook does not use any commands.
+
+## Playbook Inputs
+
+---
+
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| InstanceID | ID of the AWS EC2 instance. |  | Required |
+| Port | TCP/UDP Port to be restricted. | alert.remoteport | Required |
```
Comment (ShirleyDenkberg) [inline Packs/AWS-Enrichment-Remediation/Playbooks/AWS_-_Security_Group_Remediation_v2_README.md:30]:
```suggestion
| Port | TCP/UDP port to be restricted. | alert.remoteport | Required |
```

File: Packs/AWS-Enrichment-Remediation/README.md [hunk 9]
```diff
@@ -28,6 +29,11 @@ AWS - Security Group Remediation playbook replaces current security groups assoc
 
 ![AWS - Security Group Remediation](https://raw.githubusercontent.com/demisto/content/master/Packs/AWS-Enrichment-Remediation/doc_files/AWS_-_Security_Group_Remediation.png)
 
+#### AWS - Security Group Remediation v2
+AWS - Security Group Remediation playbook more selectively determine which security groups are over-permissive, copies them and remove only the over-permissive portions.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-Enrichment-Remediation/README.md:33]:
```suggestion
The AWS - Security Group Remediation playbook more selectively determines which security groups are over-permissive, copies them and removes only the over-permissive portions.
```

File: Packs/AWS-Enrichment-Remediation/ReleaseNotes/1_1_1.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+#### Playbooks
+##### New: AWS - Security Group Remediation v2
+Added the *AWS - Security Group Remediation v2* playbook to more selectively determine which security groups are over-permissive, copies them and remove only the over-permissive portions.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-Enrichment-Remediation/ReleaseNotes/1_1_1.md:3]:
```suggestion
Added the **AWS - Security Group Remediation v2** playbook to more selectively determine which security groups are over-permissive, copy them and remove only the over-permissive portions.
```

File: Packs/AWS-Enrichment-Remediation/ReleaseNotes/1_1_1.md [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+
+#### Scripts
+##### New: AWSRecreateSG
+Added the *AWSRecreateSG* script to determine what interface on an EC2 instance has an over-permissive security group on, determine which security groups have over-permissive rules and to replace them with a copy of the security group that has only the over-permissive portion removed.  Over-permissive is defined as sensitive ports (SSH, RDP, etc) being exposed to the internet via IPv4.
```
Comment (ShirleyDenkberg) [inline Packs/AWS-Enrichment-Remediation/ReleaseNotes/1_1_1.md:7]:
```suggestion
Added the **AWSRecreateSG** script to determine what interface on an EC2 instance has an over-permissive security group on, determine which security groups have over-permissive rules and to replace them with a copy of the security group that has only the over-permissive portion removed.  Over-permissive is defined as sensitive ports (SSH, RDP, etc) being exposed to the internet via IPv4.
```

File: Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/README.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+* aws-ec2-describe-instances
+
+## Inputs
+
+---
+
+| **Argument Name** | **Description** |
+| --- | --- |
+| instance_id | EC2 Instance ID. |
+| port | TCP/UDP Port to be restricted. |
```
Comment (ShirleyDenkberg) [inline Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/README.md:30]:
```suggestion
| port | TCP/UDP port to be restricted. |
```

**Comment by efelmandar** [comment]
@yucohen the recorded demo is enough here, finished the review, and is ready for merge.
@johnnywilkes nice work!

**Review by yucohen — APPROVED** [review]

**Pull request merged**